cluster-sim/cluster_sim/locale/ru/LC_MESSAGE/cluster_sim.po

```diff
@@ -0,0 +1,966 @@
+"\n"
+"     ?- Help, <ESC>- Back, <ENTER>- Refresh, <CTRL-L>- Event Log\n"
+"      - Помощь     - Назад        - Обновить         - Протокол\n"
+"> "
+
+#: fake\_power\_control.py:344
+msgid ""
+"\n"
+"------- Control Outlet --------------------------------------------------------\n"
+"\n"
+"        Name         : Outlet %s\n"
+"        Outlet       : %s\n"
+"        State        : %s\n"
+"\n"
+"     1- Immediate On\n"
+"     2- Immediate Off\n"
+"     3- Immediate Reboot\n"
+"     4- Delayed On\n"
+"     5- Delayed Off\n"
+"     6- Delayed Reboot\n"
+"     7- Cancel\n"
+"\n"
+"     ?- Help, <ESC>- Back, <ENTER>- Refresh, <CTRL-L>- Event Log\n"
+"> "
+msgstr ""
+"\n"
+"------- Control Outlet -- Управлять розеткой ----------------------------------\n"
+"\n"
+"        Name   Имя       : Outlet %s\n"
+"        Outlet Розетка   : %s\n"
+"        State  Состояние : %s\n"
+"\n"
+"     1- Immediate On     -- Немедленное включение\n"
+"     2- Immediate Off    -- Немедленное отключение\n"
+"     3- Immediate Reboot -- Немедленная перезагрузка\n"
+"     4- Delayed On       -- Отложенное включение\n"
+"     5- Delayed Off      -- Отложенное отключение\n"
+"     6- Delayed Reboot   -- Отложенная перезагрузка\n"
+"     7- Cancel           -- Отмена\n"
+"\n"
+"     ?- Help, <ESC>- Back, <ENTER>- Refresh, <CTRL-L>- Event Log\n"
+"      - Помощь     - Назад        - Обновить         - Протокол\n"
+"> "
+
+#: fake\_power\_control.py:367
+msgid ""
+"\n"
+"        -----------------------------------------------------------------------\n"
+"        I'm too lazy to go through all the various permutations.\n"
+"        The key line expected by fence\_apc is the next one:\n"
+"\n"
+"        Enter 'YES' to continue or <ENTER> to cancel : "
+msgstr ""
+"\n"
+"        -----------------------------------------------------------------------\n"
+"        Я слишком ленив, чтобы продираться через всевозможные перестановки.\n"
+"        Нужная строчка, которую ждёт fence\_apc, вот эта:\n"
+"\n"
+"        Введите 'YES', чтобы продолжить или <ENTER>, чтобы отменить: "
+
+#: fake\_power\_control.py:391
+msgid ""
+"%sCommand successfully issued.\r\n"
+msgstr "Команда успешно введена."
+
+#: fake\_power\_control.py:394
+msgid ""
+"\r\n"
+"%sPress <ENTER> to continue...\r\n"
+msgstr "\r\n%sНажмите <ENTER> для продолжения...\r\n"
+
+#: fake\_power\_control.py:401
+msgid ""
+"Connection Closed - Bye\r\n"
+msgstr "Соединение закрыто - до свидания\r\n"
+
+#: fake\_power\_control.py:403
+msgid "Client %s:%s logged out"
+msgstr "Клиент %s:%s вышел"
+
+#: fake\_power\_control.py:414
+msgid "%s: Running %s -> %s on %s:%s"
+msgstr "%s: Выполняю %s -> %s на %s:%s"
+
+#: fake\_power\_control.py:553
+msgid "stopping %s in poweroff\_hook"
+msgstr "останавливаю %s в poweroff\_hook"
+
+#: fake\_power\_control.py:563
+msgid "starting %s in poweron\_hook"
+msgstr "запускаю %s в poweron\_hook"
+
+#: fake\_power\_control.py:567
+msgid "starting server for %s"
+msgstr ""
```

Comment:
Does the empty `msgstr` here (and below) result in an empty string being logged?  Or even an exception due to the arguments not having `%s`(s) to substitute into?

Reply:
It should be the current "end of translation".
As I noted in the commit msg, I've translated about 1/3 of the file.
I consider this commit as a PoC to make sure I do it in an acceptable way.

Reply:
I'm not a localization expert by any stretch of the imagination so I probably don't personally have much of an opinion on how it's done.  Other than the notes I have made here already, I don't have a problem with the implementation.

One thing that might be worth considering though is if each package should have it's own set of `.po` files or there should be a single (optional even -- if possible) _localization_ package for all of IML.  That could be a bridge to cross in the future though.

Ultimately, I don't think you finishing the rest of the translations in the .po files is going to be wasted work even if we do decide to restructure the packaging somewhat.

One thing I would say though is that the cluster-simulator is probably the item least in need of localization.  I would think there is more value in localizing IML-proper.

Reply:
I've started off with the most simple-to-localize (from my PoV) chunk of code :)
It's pure python without django (which has its own set of i18n/l10n rules), where my knowledge is quite limited.

If you have no major contras on *how* it was done, I'll, probably, finish it up and then move onto the other parts.

As of separation... It's quite common to have separate language packs.
But for the routine, one will have to update automatically generated `.pot` file *within* the package anyway, while hand written `.po` files can live separately for sure.
As soon, as an IML langpack repo become available, they can be moved there.

Reply:
I have no major issues at this point.  I think it's worthwhile continuing.  I have no issues with the way the translations are being imported in python.  Even if we do some reorganisation in the future, the value of the actual translations and how the python code utilises them files remains.